packages/@ember/-internals/glimmer/lib/component.ts

```diff
@@ -727,13 +728,10 @@ const Component = CoreView.extend(
       this[ROOT_REF] = new RootReference(this);
       this[BOUNDS] = null;
 
-      // If in a tagless component, assert that no event handlers are defined
-      assert(
-        // tslint:disable-next-line:max-line-length
-        `You can not define a function that handles DOM events in the \`${this}\` tagless component since it doesn't have any DOM element.`,
-        this.tagName !== '' ||
-          !this.renderer._destinedForDOM ||
-          !(() => {
+      if (DEBUG) {
```

Comment:
```suggestion
      if (DEBUG && this.renderer._destinedForDOM) {
```

This _should_ fix the errors you are seeing in CI (around instantiating the event dispatcher in non-interactive / aka FastBoot mode).